### PR TITLE
feat: removing migrated grpc- and proto- modules from repository

### PR DESCRIPTION
**Pull request by suztomo**
The proto- and grpc- modules have been migrated to
https://github.com/googleapis/gapic-generator-java/tree/main/java-iam

### CODE REVIEW

File: .repo-metadata.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   "repo": "googleapis/java-iam",
   "repo_short": "java-iam",
   "library_type": "OTHER",
-  "distribution_name": "com.google.api.grpc:proto-google-iam-v1",
+  "distribution_name": "com.google.cloud:google-iam-policy",
```
Comment (suztomo) [inline .repo-metadata.json:14]: Memo: OwlBot Postprocessor modifies versions.txt and other files based on repo-metadata.json.

https://github.com/googleapis/synthtool/blob/7197a001ffb6d8ce7b0b9b11c280f0c536c1033a/docker/owlbot/java/src/fix-poms.py#L279

File: .kokoro/release/publish_javadoc.sh [hunk 2]
```diff
@@ -33,7 +33,7 @@ python3 -m pip install --require-hashes -r .kokoro/requirements.txt
 # compile all packages
 mvn clean install -B -q -DskipTests=true
 
-export NAME=proto-google-iam-v1
+export NAME=google-iam-policy
```
Comment (suztomo) [inline .kokoro/release/publish_javadoc.sh:36]: OwlBot PostProcessor made this change, based on the repo-metadata.json change.

File: .kokoro/release/publish_javadoc11.sh [hunk 3]
```diff
@@ -33,7 +33,7 @@ python3 -m pip install --require-hashes -r .kokoro/requirements.txt
 # compile all packages
 mvn clean install -B -q -DskipTests=true
 
-export NAME=proto-google-iam-v1
+export NAME=google-iam-policy
```
Comment (suztomo) [inline .kokoro/release/publish_javadoc11.sh:36]: OwlBot PostProcessor made this change, based on the repo-metadata.json change.

**Review by ddixit14 — APPROVED** [review]

**Comment by suztomo** [comment]
It's "feat" to make it releasable unit with a minor version bump.

**Pull request merged**